qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -369,13 +435,28 @@ def _score_heuristic(self, heuristic, front_layer, extended_set, layout, swap_qu
 
         raise TranspilerError("Heuristic %s not recognized." % heuristic)
 
+    def _undo_operations(self, operations, dag, layout):
+        """Mutate ``dag`` and ``layout`` by undoing the swap gates listed in ``operations``."""
+        if dag is None:
+            for operation in reversed(operations):
+                layout.swap(*operation.qargs)
+        else:
+            for operation in reversed(operations):
+                dag.remove_op_node(operation)
+                p0 = self._bit_indices[operation.qargs[0]]
+                p1 = self._bit_indices[operation.qargs[1]]
+                layout.swap(p0, p1)
+
 
 def _transform_gate_for_layout(op_node, layout, device_qreg):
     """Return node implementing a virtual op on given layout."""
     mapped_op_node = copy(op_node)
+    mapped_op_node.qargs = [device_qreg[layout._v2p[x]] for x in op_node.qargs]
+    return mapped_op_node
 
-    premap_qargs = op_node.qargs
-    mapped_qargs = map(lambda x: device_qreg[layout._v2p[x]], premap_qargs)
-    mapped_op_node.qargs = list(mapped_qargs)
 
-    return mapped_op_node
+def _shortest_swap_path(target, coupling_map, layout):
+    v_start, v_goal = target
+    start, goal = layout._v2p[v_start], layout._v2p[v_goal]
+    for swap in list(retworkx.dijkstra_shortest_paths(coupling_map.graph, start)[goal])[1:-1]:
```

Comment:
You can add a target kwarg to the dijkstra call to return early if we reach the goal node. It still returns a dict in this case but with a single entry for the specified target node:

```suggestion
    for swap in list(retworkx.dijkstra_shortest_paths(coupling_map.graph, start, target=goal)[goal])[1:-1]:
```

This likely won't make a noticeable difference but in edge cases it could improve performance

Reply:
Good catch, thanks - I thought I'd already put that in there!  Fixed in 5b238ee07.